db/migrate/20200616162646_move_all_calculators_outside_the_spree_namespace.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class MoveAllCalculatorsOutsideTheSpreeNamespace < ActiveRecord::Migration
+  def up
+    convert_calculator("DefaultTax")
+    convert_calculator("FlatPercentItemTotal")
+    convert_calculator("FlatRate")
+    convert_calculator("FlexiRate")
+    convert_calculator("PerItem")
+    convert_calculator("PriceSack")
+  end
+
+  def down
+    revert_calculator("DefaultTax")
+    revert_calculator("FlatPercentItemTotal")
+    revert_calculator("FlatRate")
+    revert_calculator("FlexiRate")
+    revert_calculator("PerItem")
+    revert_calculator("PriceSack")
+  end
+
+  private
+
+  def convert_calculator(calculator_base_name)
+    update_calculator("Spree::Calculator::" + calculator_base_name,
+                      "Calculator::" + calculator_base_name)
+  end
+
+  def revert_calculator(calculator_base_name)
+    update_calculator("Calculator::" + calculator_base_name,
+                      "Spree::Calculator::" + calculator_base_name)
+  end
+
+  def update_calculator(from, to)
+    Spree::Calculator.connection.execute(
+      "UPDATE spree_calculators SET type = '" + to + "' WHERE type = '" + from + "'"
+    )
+  end
```

Comment:
I'm just wondering. Why do we do it with plain SQL? do you want to skip updating the timestamps and going through the validations? In that case, AR allows for that and IMO the intent is more clear but that's a minor thing that felt worth sharing.

Pretty good migration :ok_hand:

Reply:
I definitely don't trust spree code for callbacks and magic, hence SQL :-)
But now I am curious how it would look like with AR. Can you share an example?

Reply:
Fair enough but :hankey: why did I ask :trollface:

Reply:
https://apidock.com/rails/v4.0.2/ActiveRecord/Persistence/update_columns is what you need.

Reply:
I am sorry for giving you more work, It's just an opportunity for me to learn.
I see what you mean! thanks.
I could change it, but if you dont mind I will go as is, just to save the time of converting things and testing.

Reply:
Yes, don't worry I was just kidding. You know I like sharing this know-how